src/components/Button/Button.tsx

```diff
@@ -19,44 +15,14 @@ export interface Props {
    * Disables the field and prevents editing the contents
    */
   disabled?: boolean;
-  /**
-   * Toggles button that fills the full width of its container
-   */
-  fullWidth?: boolean;
-  /**
-   * Visually hide button text (but text is still accessible to assistive technology)
-   */
-  hideText?: boolean;
   /**
    * Link to URL. If href is present, the button will be rendered as an <a> element.
    */
   href?: string;
-  /**
-   * Name of SVG icon (i.e. caret-down, minus, warning)
-   */
-  iconName?: string;
-  /**
-   * Determines position of icon relative to button text.
-   * - **before** places icon before button text
-   * - **after** places icon after button text
-   */
-  iconPosition?: 'before' | 'after';
```

Comment:
We considered using a similar API as well and decided against it. The reason why is because all of the icons are in square bounding boxes, but the icons themselves do not always have a square ratio, which means some skinny icons will have more space between the icon and the button text than others. So we decided to let devs pass in their own icons which would allow them to customize the space between the icon and the text. That being said, I'm open to the suggestion that I'm overthinking this and being nitpicky and we should just use this API because it's easier for devs. Here's the conversation where we discussed passing in icons via children vs using props: https://github.com/chanzuckerberg/edu-design-system/pull/643#discussion_r713295633

Reply:
@dierat no this is an important architectural convention! And it's something that me and @anniehu4 touched on briefly. Icons is its own can of worms and ought to have some careful consideration.

We should talk about this as a broader group (including design), but it really comes down to control. Ceding control at a `Button` component level and giving devs the ability to pass in their own icons and control spacing means that each button needs design QA for every instance vs addressing it at the EDS level. There are no right answers here! I think we'll flag this with the broader group to get a sense of what design's expectations are, and then marry it with the technical considerations and requirements.

Reply:
I'll revert this change and ask design whether we want to support this feature in this component: https://app.shortcut.com/czi-edu/story/188471/button-api-questions-for-design